learning_journal/tests/tests.py

```diff
@@ -1,65 +1,169 @@
+    request.addfinalizer(teardown)
+    return config
 
-        self.session = get_tm_session(session_factory, transaction.manager)
 
-    def init_database(self):
-        from .models.meta import Base
-        Base.metadata.create_all(self.engine)
+@pytest.fixture()
+def database_session(configuration, request):
+    """Create a session to interact with the testing database."""
+    SessonFactory = configuration.registry['dbsession_factory']
+    session = SessonFactory()
+    engine = session.bind
+    Base.metadata.create_all(engine)
 
-    def tearDown(self):
-        from .models.meta import Base
+    def teardown():
+        session.transaction.rollback()
+
+    request.addfinalizer(teardown)
+    return session
+
+
+@pytest.fixture()
+def dummy_request(database_session):
+    """Instantiate a HTTP request with database session for each new request."""
+    return testing.DummyRequest(dbsession=database_session)
+
+
+@pytest.fixture()
+def add_models(dummy_request):
+    """Add the generated model to the database."""
+    dummy_request.dbsession.add_all(POSTS)
+
+Fake = faker.Faker()
+
+POSTS = [MyModel(
+    title=Fake.text(20),
+    id=i,
+    body=Fake.text(100),
+    creation_date=datetime.datetime.now(),
+) for i in range(20)]
+
+
+# Unit Tests
+
+def test_add_post(database_session):
+    """Test if database entries are added."""
+    database_session.add_all(POSTS)
+    query = database_session.query(MyModel).all()
+    print('database contents: ', database_session.query(MyModel).all())
+    assert len(query) == len(POSTS)
+
+
+def test_list_return(dummy_request, add_models):
+    """Test that the list view returns 20 entries."""
+    from learning_journal.views.default import home_page
+    view = home_page(dummy_request)
+    assert len(view["entries"]) == 20
+
+
+def test_empty_list(dummy_request):
+    """Test that the list view is empty when no entries are added."""
+    from learning_journal.views.default import home_page
+    view = home_page(dummy_request)
+    assert len(view["entries"]) == 0
+
+
+def test_correct_entries(database_session, dummy_request, add_models):
+    """Test entries are added in the right order with expected content."""
+    from learning_journal.views.default import home_page
+    home_page(dummy_request)
+    query = database_session.query(MyModel).all()
+    assert query[0].title == POSTS[0].title
+
+
+def test_detail_view_content(database_session, dummy_request, add_models):
+    """Test that database ids match main list ids."""
+    from learning_journal.views.default import detail_page
+    dummy_request.matchdict['id'] = 1
+    detail_page(dummy_request)
+    query = database_session.query(MyModel).all()
+    assert query[0].title == POSTS[0].title
+
+
+def test_edit(database_session, dummy_request, add_models):
+    """Test database items can be edited."""
+    query = database_session.query(MyModel).first()
+    query.title = 'edited title'
+    assert query.title == 'edited title'
```

Comment:
This test isn't actually testing your edit function.  You are just re-assigning the value stored in the title attribute of the query object.